### PR TITLE
session: select the fastest probe service

### PR DESCRIPTION
**Pull request by bassosimone**
Part of https://github.com/ooni/probe-engine/issues/407

Includes fixes for a flaky integration test that otherwise prevented clean merging

### CODE REVIEW

File: oonimkall/task_test.go [hunk 1]
```diff
@@ -431,7 +431,7 @@ func TestIntegrationInterruptNdt7(t *testing.T) {
 		t.Fatal(err)
 	}
 	go func() {
-		<-time.After(7 * time.Second)
+		<-time.After(11 * time.Second)
```
Comment (bassosimone) [inline oonimkall/task_test.go:434]: This seems to fix an integration test that started failing with GitHub Actions (but not locally)

File: oonimkall/task_test.go [hunk 2]
```diff
@@ -462,8 +462,8 @@ func TestIntegrationInterruptNdt7(t *testing.T) {
 		"status.end",
 		"task_terminated",
 	}
-	if !reflect.DeepEqual(keys, expect) {
-		t.Fatal("seen different keys than expected")
+	if diff := cmp.Diff(expect, keys); diff != "" {
+		t.Fatalf(diff)
```
Comment (bassosimone) [inline oonimkall/task_test.go:466]: This has been useful to understand why an integration test was failing

File: oonimkall/task_test.go [hunk 3]
```diff
@@ -411,8 +411,8 @@ func TestIntegrationInterruptExampleWithInput(t *testing.T) {
 		"status.end",
 		"task_terminated",
 	}
-	if !reflect.DeepEqual(keys, expect) {
-		t.Fatalf("seen different keys than expected: %+v", keys)
+	if diff := cmp.Diff(expect, keys); diff != "" {
+		t.Fatalf(diff)
```
Comment (bassosimone) [inline oonimkall/task_test.go:415]: This has been done to sync this test with another test that we changed to investigate failures

File: probeservices/client_test.go [hunk 4]
```diff
@@ -204,3 +206,385 @@ func TestSortEndpoints(t *testing.T) {
...
+	// put onion first so we also verify that we sort the endpoints
+	in := []model.Service{{
+		Type:    "onion",
+		Address: "httpo://jehhrikjjqrlpufu.onion",
+	}, {
+		Type:    "https",
+		Address: "https://ams-ps.ooni.io",
+	}, {
+		Type:    "https",
+		Address: "https://hkg-ps.ooni.io",
```
Comment (hellais) [inline probeservices/client_test.go:255]: I would suggest replacing these with another string that is clearly not used, as these domain names are very similar to one that exists, but in fact they do not.

Comment (bassosimone) [reply probeservices/client_test.go:255]: https://github.com/ooni/probe-engine/issues/664

**Pull request merged**